master/internal/rm/kubernetesrm/pods.go

```diff
@@ -1568,7 +1568,11 @@ func (p *pods) getNonDetPods() []k8sV1.Pod {
 		return nonDetPods
 	}
 	for _, p := range pList.Items {
-		if _, ok := p.Labels["determined"]; !ok {
+		_, ok1 := p.Labels[determinedLabel]
+		_, ok2 := p.Labels[determinedSystemLabel]
+		_, ok3 := p.Labels[determinedPreemptionLabel]
+
+		if !(ok1 || ok2 || ok3) {
 			if p.Spec.NodeName != "" {
```

Comment:
Why would the node name be `""`? i.e., what are the conditions that might lead to that situation?

Reply:
I think if we query the API server for pods that exist in a cluster but are not yet scheduled on a node, then the pod won't yet have a `nodeName`

Reply:
Makes sense - thanks Amanda!